src/schema/schema_alter.c

```diff
@@ -63,7 +54,164 @@ err:	__wt_free(session, config);
 	if (ret == WT_NOTFOUND)
 		ret = ENOENT;
 
-	WT_TRET(__wt_meta_track_off(session, true, ret != 0));
+	return (ret);
+}
+
+/*
+ * __alter_file --
+ *	Alter a file.
+ */
+static int
+__alter_file(WT_SESSION_IMPL *session, const char *cfg[])
+{
+	WT_DECL_RET;
+	const char *filename, *uri;
+
+	/*
+	 * We know that we have exclusive access to the file.  So it will be
+	 * closed after we're done with it and the next open will see the
+	 * updated metadata.
+	 */
+	filename = uri = session->dhandle->name;
+	if (!WT_PREFIX_SKIP(filename, "file:"))
+		return (__wt_unexpected_object_type(session, uri, "file:"));
+
+	WT_RET(__wt_meta_track_on(session));
 
+	WT_ERR(__alter_apply(session,
+	    uri, cfg, WT_CONFIG_BASE(session, file_meta)));
+
+err:	WT_TRET(__wt_meta_track_off(session, true, ret != 0));
 	return (ret);
 }
+
+/*
+ * __alter_tree --
+ *	Alter an index or colgroup reference.
+ */
+static int
+__alter_tree(WT_SESSION_IMPL *session, const char *name, const char *cfg[])
+{
+	WT_CONFIG_ITEM cval;
+	WT_DECL_ITEM(data_source);
+	WT_DECL_RET;
+	char *value;
+	bool is_colgroup;
+
+	value = NULL;
+
+	is_colgroup = WT_PREFIX_MATCH(name, "colgroup:");
+	if (!is_colgroup && !WT_PREFIX_MATCH(name, "index:"))
+		WT_ERR_MSG(session, EINVAL,
+		    "expected a 'colgroup:' or 'index:' source: '%s'", name);
```

Comment:
It feels like alter should use consistent error reporting. i.e: either this or the `__wt_unexepected_object_type` used in `__alter_file`

Reply:
+1

Reply:
fixed